### PR TITLE
Switch to version catalog for dependencies

### PR DESCRIPTION
**Pull request by bulldog98**
This PR changes the build according to #234

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -1,36 +1,5 @@
...
-equalsverifierVersion=3.10.1
-assertJVersion=3.23.1
-mockitoVersion=4.7.0
-jsr305Version=3.0.2
-checkstyleVersion=10.3.2
-# Libraries
-apiGuardianVersion=1.1.2
 # MC
 minecraftVersion=1.19.2
 parchmentVersion=2022.08.21
```
Comment (raoulvdberge) [inline gradle.properties]: Can these be removed now somehow?

File: refinedstorage2-core-api/build.gradle [hunk 2]
```diff
@@ -1,10 +1,8 @@
 archivesBaseName = 'refinedstorage2-core-api'
 
 dependencies {
-    testImplementation "org.junit.jupiter:junit-jupiter-api:${project.junitVersion}"
-    testRuntimeOnly "org.junit.jupiter:junit-jupiter-engine:${project.junitVersion}"
-    testImplementation "org.junit.jupiter:junit-jupiter-params:${project.junitVersion}"
-    testImplementation "org.assertj:assertj-core:${project.assertJVersion}"
+    testImplementation testLibs.bundles.junit
```
Comment (raoulvdberge) [inline refinedstorage2-core-api/build.gradle]: I'm not really a fan of using bundles for junit. The engine should only be provided at runtime, not in implementation.

**Review by raoulvdberge — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 1 above)
Comment (bulldog98) [reply gradle.properties]: At the moment I was not able to configure the parchment version via version catalog, opened a bug in the fabric loom gradle plugin about that https://github.com/FabricMC/fabric-loom/issues/782

File: refinedstorage2-core-api/build.gradle (hunk 2 above)
Comment (bulldog98) [reply refinedstorage2-core-api/build.gradle]: Will fix that today or tomorrow.

File: gradle.properties (hunk 1 above)
Comment (raoulvdberge) [reply gradle.properties]: In that case, maybe put a TODO comment on there with a link to your issue.

Comment (bulldog98) [reply gradle.properties]: I had a thought, how to get rid of the versions there, you can also only define a version in the catalog and use it in a string interpolation, we should still add that TODO with the link. Not sure if it should be at the position where the version is defined, or at the position where it's used.

Comment (raoulvdberge) [reply gradle.properties]: That's even better. Probably in the position where it's used,  I think..?

**Comment by raoulvdberge** [comment]
Can you also rebase? I did a minor change to the checkstyle config on develop.

File: refinedstorage2-core-api/build.gradle (hunk 2 above)
Comment (bulldog98) [reply refinedstorage2-core-api/build.gradle]: Done

File: gradle.properties (hunk 1 above)
Comment (bulldog98) [reply gradle.properties]: Could you check if that is ok like that?

**Comment by raoulvdberge** [comment]
The build is failing. `refinedstorage2-network-test` needs to have Junit on the implementation scope.

**Comment by bulldog98** [comment]
> The build is failing. `refinedstorage2-network-test` needs to have Junit on the implementation scope.
Fixed that

File: gradlew [hunk 3]
```diff
@@ -32,10 +32,10 @@
 #       Busybox and similar reduced shells will NOT work, because this script
 #       requires all of these POSIX shell features:
 #         * functions;
-#         * expansions �$var�, �${var}�, �${var:-default}�, �${var+SET}�,
-#           �${var#prefix}�, �${var%suffix}�, and �$( cmd )�;
-#         * compound commands having a testable exit status, especially �case�;
-#         * various built-in commands including �command�, �set�, and �ulimit�.
+#         * expansions «$var», «${var}», «${var:-default}», «${var+SET}»,
```
Comment (raoulvdberge) [inline gradlew:35]: Could you undo changes in this file?

File: refinedstorage2-platform-forge/build.gradle [hunk 4]
```diff
@@ -4,9 +4,10 @@ buildscript {
             url = 'https://maven.minecraftforge.net'
         }
     }
-    dependencies {
-        classpath group: 'net.minecraftforge.gradle', name: 'ForgeGradle', version: '5.1.+', changing: true
```
Comment (raoulvdberge) [inline refinedstorage2-platform-forge/build.gradle:8]: Why did you remove this?

File: refinedstorage2-platform-forge/build.gradle [hunk 5]
```diff
@@ -20,12 +21,10 @@ repositories {
     }
 }
 
-apply plugin: 'net.minecraftforge.gradle'
```
Comment (raoulvdberge) [inline refinedstorage2-platform-forge/build.gradle:23]: Why did you remove this?

**Review by raoulvdberge — CHANGES_REQUESTED** [review]
Almost there. Just a small question and request.

File: refinedstorage2-platform-forge/build.gradle (hunk 4 above)
Comment (bulldog98) [reply refinedstorage2-platform-forge/build.gradle:8]: Ah that was the old way to apply gradle plugins, that is the legacy way to add a plugin see https://docs.gradle.org/current/userguide/plugins.html#sec:old_plugin_application

File: refinedstorage2-platform-forge/build.gradle [hunk 6]
```diff
@@ -4,9 +4,10 @@ buildscript {
             url = 'https://maven.minecraftforge.net'
         }
     }
-    dependencies {
-        classpath group: 'net.minecraftforge.gradle', name: 'ForgeGradle', version: '5.1.+', changing: true
-    }
+}
+
+plugins {
+    alias(forge.plugins.forgeGradle)
```
Comment (bulldog98) [inline refinedstorage2-platform-forge/build.gradle:10]: This is the modern and recommended way to use a plugin

Comment (raoulvdberge) [reply refinedstorage2-platform-forge/build.gradle:10]: cool, thanks!

File: gradlew (hunk 3 above)
Comment (raoulvdberge) [reply gradlew:35]: Heh. Looks like github was just rendering it weirdly.
image

Checked the branch out and looks fine locally.

**Review by raoulvdberge — APPROVED** [review]

**Comment by raoulvdberge** [comment]
Thanks!

**Pull request merged**